test/functional/hosts_controller_test.rb

```diff
@@ -505,11 +505,22 @@ def set_remote_user_to(user)
       @host1, @host2 = FactoryGirl.create_list(:host, 2, :managed)
     end
 
-    test 'build' do
+    test 'build without reboot' do
       assert !@host1.build
       assert !@host2.build
       multiple_hosts_submit_request('build', [@host1.id, @host2.id],
-                                    'The selected hosts will execute a build operation on next reboot')
+                                    'The selected hosts will execute a build operation on next reboot',
+                                    {:host => { :build => 0 }})
+      assert Host.find(@host1).build
+      assert Host.find(@host2).build
+    end
+
+    test 'build with reboot' do
```

Comment:
This test needs to check that `host.power.reboot` was called.  I think it's probably returning false for `supports_power_and_running` in the test so is skipping the step, so I'd suggest adding a `Host::Managed.any_instance.expects(:supports_power_and_running).twice.returns(true)` or similar, then expect the `.reboot` call too.